types/mydecimal.go

```diff
@@ -391,9 +391,8 @@ func (d *MyDecimal) FromString(str []byte) error {
 	if innerIdx != 0 {
 		d.wordBuf[wordIdx] = word * powers10[digitsPerWord-innerIdx]
 	}
-	if endIdx+1 < len(str) && (str[endIdx] == 'e' || str[endIdx] == 'E') {
-		exponent, err1 := strToInt(string(str[endIdx+1:]))
-		// TODO: need a way to check if there is at least one digit.
+	if endIdx+1 <= len(str) && (str[endIdx] == 'e' || str[endIdx] == 'E') {
+		exponent, err1 := strconv.ParseInt(string(str[endIdx+1:]), 10, 64)
```

Comment:
If the exponent part is `e2e`, then we the exponent should be `2`, instead of `0`.